sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -359,11 +359,12 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         (None, message)
 
       // our bucketing is un-compatible with hive(different hash function)
-      case _ if table.bucketSpec.nonEmpty =>
+      case Some(serde) if table.bucketSpec.nonEmpty =>
         val message =
           s"Persisting bucketed data source table $qualifiedTableName into " +
-            "Hive metastore in Spark SQL specific format, which is NOT compatible with Hive. "
-        (None, message)
+            "Hive metastore in Spark SQL specific format, which is NOT compatible with Hive. " +
+            "But Hive can read it as not bucketed table."
+        (Some(newHiveCompatibleMetastoreTable(serde)), message)
```

Comment:
Should we set `bucketSpec = None`?

Reply:
Not necessary:
https://github.com/apache/spark/blob/fee695d0cf211e4119c7df7a984708628dc9368a/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L1009-L1023